### PR TITLE
Add cycle detection (#7532)

### PR DESCRIPTION
**Pull request by ethan92429**
Fix for https://github.com/OpenAPITools/openapi-generator/issues/7532

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by ethan92429** [comment]
Sorry - this is still a WIP. Looking at the samples

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonExperimentalClientCodegen.java [hunk 1]
```diff
@@ -1329,9 +1329,14 @@ private MappedModel getDiscriminatorMappedModel(CodegenDiscriminator disc) {
      *                    ModelName( line 0
      *                        some_property='some_property_example' line 1
      *                    ) line 2
+     * @param includedSchemas schemas already included in this example - for cycle detection.
      * @return the string example
      */
-    private String toExampleValueRecursive(String modelName, Schema schema, Object objExample, int indentationLevel, String prefix, Integer exampleLine) {
+    private String toExampleValueRecursive(String modelName, Schema schema, Object objExample, int indentationLevel, String prefix, Integer exampleLine, List<Schema> includedSchemas) {
+        if (includedSchemas.stream().filter(s->schema.equals(s)).count() > 1) {
+            return "";
```
Comment (spacether) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonExperimentalClientCodegen.java:1337]: How about defining this as a variable
```
boolean cycleFound = includedSchemas.stream().filter(s->schema.equals(s)).count() > 1;
if (cycleFound) {
  return "";
}
```

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonExperimentalClientCodegen.java [hunk 2]
```diff
@@ -1329,9 +1329,14 @@ private MappedModel getDiscriminatorMappedModel(CodegenDiscriminator disc) {
      *                    ModelName( line 0
      *                        some_property='some_property_example' line 1
      *                    ) line 2
+     * @param includedSchemas schemas already included in this example - for cycle detection.
```
Comment (spacether) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonExperimentalClientCodegen.java:1332]: Can you further describe includedSchemas here?
includedSchemas are a list of schemas that we have moved through to get here. If the new schemas that we are looking at is in includedSchemas then we have hit a cycle.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonExperimentalClientCodegen.java (hunk 1 above)
Comment (spacether) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonExperimentalClientCodegen.java:1339]: Thanks!

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/python/PythonClientTest.java [hunk 3]
```diff
@@ -494,7 +494,7 @@ public void testRecursiveExampleValueWithCycle() throws Exception {
         final OpenAPI openAPI = TestUtils.parseFlattenSpec("src/test/resources/3_0/issue_7532.yaml");
         final PythonExperimentalClientCodegen codegen = new PythonExperimentalClientCodegen();
         codegen.setOpenAPI(openAPI);
-        Schema tree = openAPI.getComponents().getSchemas().get("Tree");
+        Schema tree = openAPI.getComponents().getSchemas().get("Forest");
```
Comment (spacether) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/python/PythonClientTest.java:497]: How about a comment mentioning that this node has a cycle.
Or naming it something like schemaWithACycleInAllOf?

**Comment by spacether** [comment]
@ethan92429 can you update the samples? I need them updated before I can approve and merge it. Thanks!

**Comment by ethan92429** [comment]
Done! 😄

**Comment by spacether** [comment]
Bitrise error looks unrelated

**Review by spacether — APPROVED** [review]
Looks good, thank you for your PR!

**Pull request merged**